### DISCUSSION TITLE
How to avoid of chat main example stops abruptly?

### DISCUSSION THREAD

**Discussion by Zapotecatl** [ggml-org/llama.cpp#7180] [Q&A]
Hi,

In the main example there is a part where if an error occurs the chat ends:

```
LOG("eval: %s\n", LOG_TOKENS_TOSTR_PRETTY(ctx, embd).c_str());

if (llama_decode(ctx, llama_batch_get_one(&embd[i], n_eval, n_past, 0))) {
     LOG_TEE("%s : failed to eval\n", __func__);
     return 1;
}
```
I don't understand in which cases the error occurs and how to avoid it. The parameters I use in my command are this:

`-m C:\\model\\llama2_7b_chat_uncensored.Q4_K_M.gguf -f C:\\model\\chat-with-bob.txt -c 2048 -b 2048 -n 64 --keep -1 --temp 0.9 -- repeat_penalty 1.1 -i -r "User:" --log-disable`

At the moment I am using a work around in which if the error occurs within an external while(true) the chat is launched again and it gives the impression that the chat continues and the user does not perceive that the program ended abruptly.

So, I'm wondering if Is there a solution to ensure that the error never occurs to avoid my ugly work around?

Thanks for the help!

**Comment by marcingomulkiewicz** [comment]
If I see it right, you're setting -n (--n-predict) to 64 - which means the server is _supposed_ to stop generating after 64 tokens. You can set it to a larger value (the same problem, just different threshold), or -1: this is IIRC unlimited, of course, there's risk with unlimited - some models for some prompts can start babling nonsense or even loop.

**Reply by Zapotecatl** [reply]
Hi @marcingomulkiewicz ,

Thanks for your replay. Yes, I had previously tried setting it to -1, but the error kept happening. @MaggotHATE wrote in another discussion: _"I have finally seen the effect of this particular error: indeed, it happens when the size of input is greater than the model's context size."_ 

https://github.com/MaggotHATE/Llama_chat/issues/2#issuecomment-2117194283

(Although I haven't tried it yet in Llama 2 to see if it happens when I enter very large prompts. When I do the test I will come back here.)
